File: Algorithms/Bit Manipulation/The Great XOR.py
import os

#
# Complete the 'theGreatXor' function below.
#
# The function is expected to return a LONG_INTEGER.
# The function accepts LONG_INTEGER x as parameter.
#

def theGreatXor(x):
    # Write your code here
    result = 0
    bit_position = 0

    while x > 0:
        if x & 1 == 0:
            result += (1 << bit_position)
        bit_position += 1
        x >>= 1

    return result


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    q = int(input().strip())

    for q_itr in range(q):
        x = int(input().strip())

        result = theGreatXor(x)

        fptr.write(str(result) + '\n')

    fptr.close()

File: Algorithms/Bit Manipulation/The Great XOR.php
<?php

/*
 * Complete the 'theGreatXor' function below.
 *
 * The function is expected to return a LONG_INTEGER.
 * The function accepts LONG_INTEGER x as parameter.
 */

function theGreatXor($x) {
    # Write your code here
    $result = 0;
    $bit_position = 0;

    while ($x > 0) {
        if (($x & 1) == 0) {
            $result += (1 << $bit_position);
        }
        $bit_position++;
        $x >>= 1;
    }

    return $result;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$q = intval(trim(fgets(STDIN)));

for ($q_itr = 0; $q_itr < $q; $q_itr++) {
    $x = intval(trim(fgets(STDIN)));

    $result = theGreatXor($x);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
